srai/embedders/base.py

```diff
@@ -0,0 +1,24 @@
+"""Base class for embedders."""
+
+import abc
+from typing import Any
+
+import geopandas as gpd
+import numpy as np
+import numpy.typing as npt
+
+
+class BaseEmbedder(abc.ABC):
+    """Base abstract class for embedders."""
+
+    @abc.abstractmethod
+    def embed(self, gdf: gpd.GeoDataFrame, **kwargs: Any) -> npt.NDArray[np.float64]:
```

Comment:
Should we lock ourselves to NumPy arrays only? Scikit recently allowed for a pandas data frame output. When returning a numeric vector we have to provide feature names that go with it (eg Count vectorizer and object categories/tags).
I don't see a reason to stay locked in NumPy since you can always run df.to_numpy().

Reply:
```suggestion
    def embed(self, gdf: gpd.GeoDataFrame) -> npt.NDArray[np.float64]:
```